### PR TITLE
dev-python/ipython: Version bump to 5.1.0, Fix python_test()

### PR DESCRIPTION
**Pull request by marbre**
@SoapGentoo 

The virtx stuff blocked the python_test() phase from exiting properly. The original comment in the python_test() phase mentioned that tests fail with python3.4. However, the with python-3.4.3-r1, the tests pass. Same for python2_7. Therefore, I removed the virtx stuff.

Furthermore cythonmagic is removed from the source.

### CODE REVIEW

File: dev-python/ipython/ipython-5.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{3,4,5} )
```
Comment (SoapGentoo) [inline dev-python/ipython/ipython-5.1.0.ebuild]: can we remove py3.3?

**Comment by marbre** [comment]
@SoapGentoo Dropped it.

**Comment by floppym** [comment]
@SoapGentoo @marbre Please do NOT drop python3.3 from packages that already have it. That fucks wit h the dependency graph and causes some real problem.

**Comment by SoapGentoo** [comment]
@floppym I thought it's okay to drop py3.3 support for new ebuilds/version bumps, just not for removing it from existing ebuilds.

**Comment by floppym** [comment]
@SoapGentoo It can be excluded from new packages. But for bumps of an existing package, it should be retained.

**Comment by floppym** [comment]
@SoapGentoo For an example of the depgraph breakage, see the portage output below. I fixed this in the tree by re-adding python3_3.

```
WARNING: One or more updates/rebuilds have been skipped due to a dependency conflict:

dev-python/ipython:0

  (dev-python/ipython-5.1.0:0/0::gentoo, ebuild scheduled for merge) conflicts with
    dev-python/ipython[python_targets_python2_7(-),python_targets_python3_3(-),python_targets_python3_4(-),python_targets_python3_5(-),-python_single_target_python2_7(-),-python_single_target_python3_3(-),-python_single_target_python3_4(-),-python_single_target_python3_5(-)] required by (dev-python/qtconsole-4.2.1:0/0::gentoo, installed)
                                                                                                                                                                                                        
    >=dev-python/ipython-4.0.0[python_targets_python2_7(-),python_targets_python3_3(-),python_targets_python3_4(-),python_targets_python3_5(-),-python_single_target_python2_7(-),-python_single_target_python3_3(-),-python_single_target_python3_4(-),-python_single_target_python3_5(-)] required by (dev-python/ipykernel-4.3.1:0/0::gentoo, installed)
                                                                                                                                                                                                        
    >=dev-python/ipython-4.0.0[python_targets_python2_7(-),python_targets_python3_3(-),python_targets_python3_4(-),python_targets_python3_5(-),-python_single_target_python2_7(-),-python_single_target_python3_3(-),-python_single_target_python3_4(-),-python_single_target_python3_5(-)] required by (dev-python/ipyparallel-5.1.1:0/0::gentoo, installed)
```

**Comment by marbre** [comment]
@SoapGentoo @floppym Actually, we get in some trouble with jupyter/ipython anyway. Especially version bumps of jupyter/ipython packages pull in new new dependencies regularly. E.g. nbconvert-4.2.0 now requires nbsphinx, which was added (by me) without python3_3 support as it was a new package.

One my major problems is that the `setup.py` files provided by the jupyter/ipython subprojects often do not contain version information regarding the requirements. Testing which version of a package is required is not always possible and would create a too large overhead for me. Thus, we need an agreement how to proceed here, because either we try to support python3_3 within all packages required for jupyter/ipython or we wait for bug reports if some dependencies do not match.

BTW, ipython-5.x is the last version  that supports python2_7.

**Comment by SoapGentoo** [comment]
@floppym I've corrected all ebuilds now:
```
* 0d8715d - dev-python/BitVector: restore python 3.3 - David Seifert
* 3c521f3 - dev-python/kivy-garden: restore python 3.3 - David Seifert
* 5d9d74a - dev-python/ipykernel: restore python 3.3 - David Seifert
* 9ec00e7 - dev-python/nbformat: restore python 3.3 - David Seifert
* 92d67e6 - dev-python/elasticsearch-py: restore python 3.3 - David Seifert
* 8af2177 - dev-python/line_profiler: restore python 3.3 - David Seifert
```
Hope that makes you a bit happier again :smile:

**Comment by floppym** [comment]
Thanks!

**Pull request closed without merging**